app/jobs/irs_attempts_events_batch_job.rb

```diff
@@ -14,17 +14,32 @@ def perform(timestamp: Time.zone.now - 1.hour, dir_path: './attempts_api_output'
       data: event_values, timestamp: timestamp, public_key: pub_key,
     )
 
-    # write to a file and store on the disk until S3 is setup
-    FileUtils.mkdir_p(dir_path)
+    # Write the file to S3 - Can we skip the file write step? Do we need to write out a temp file?
+    if IdentityConfig.store.irs_attempt_api_bucket_name.nil?
+      # write to a file and store on the disk until S3 is setup
+      FileUtils.mkdir_p(dir_path)
 
-    file_path = "#{dir_path}/#{result.filename}"
+      file_path = "#{dir_path}/#{result.filename}"
 
-    File.open(file_path, 'wb') do |file|
-      file.write(result.encrypted_data)
-    end
+      File.open(file_path, 'wb') do |file|
+        file.write(result.encrypted_data)
+      end
+      return { encryptor_result: result, file_path: file_path }
+    else
+      bucket_name = IdentityConfig.store.irs_attempt_api_bucket_name
+      bucket_url = "s3://#{bucket_name}/#{result.filename}"
+
+      puts "uploading to #{bucket_url}"
 
-    return { encryptor_result: result, file_path: file_path }
+      aws_object = Aws::S3::Resource.new.bucket(bucket_name).object(result.filename)
+      aws_object.put(body: result.encrypted_data, acl: 'private', content_type: 'text/plain')
 
-    # Write the file to S3 instead of whatever dir_path winds up being
+      puts "upload completed to #{bucket_url}"
+
+      IrsAttemptApiLogFile.create(
```

Comment:
I would not bother with creating a database table to track these? they have the timestamp in the name so we can always do point-lookups for the files if needed with a prefix search by `formatted_time` ?

Reply:
I think we do need a table. The files are useless if we don't persist the IV and encrypted key.

In the controller, we'll want to look up the file by timestamp, get this data, and then generate a presigned link for the file.